### PR TITLE
Home work4

### PR DESCRIPTION
**Pull request by Tryxman**
I'm sorry for the late with this homework 
But something with my mind , homework for me  is so difficult

### CODE REVIEW

File: Hometask4.playground/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
+//Создать словарь, где ключ - название города/страны, значение - интересный факт и континент, где распорожен город/страна (подумайте, какой тип данных лучше использовать для значения).
+
+import Foundation
+
+var worldPlaceDictionary = ["Russia": ("The biggest country in the world", "Eurasia"),
+                            "Portugal": ("One of the most safety country in the world","Europe"),
+                            "Australia": ("The same as Europe but without migrants from near East","Australia"),
```
Comment (darya-gurinovich) [inline Hometask4.playground/Contents.swift]: Please, add a space after `,` here

File: Hometask4.playground/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+worldPlaceDictionary["Russia"]    = ("Putinka", "Eurasia")
+worldPlaceDictionary["Australia"] = ("A1", "Australia") // change elements in dictionary
+
+worldPlaceDictionary.removeValue(forKey: "Minsk")
+worldPlaceDictionary["Netherlands"] = nil // remove value from the Dictionary
+
+//2. Создать отдельный словарь с названием городов/стран с длиной более 6 букв из первоначального словаря.
+
+var dictCitiesLongName = [String: (String, String)] ()
```
Comment (darya-gurinovich) [inline Hometask4.playground/Contents.swift]: It's better to use full words in the names and I believe we need to swap the order of the words: `longNameCitiesDictionary`

File: Hometask4.playground/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+dictCitiesLongName = worldPlaceDictionary.filter({ $0.key.count > 6 })
```
Comment (darya-gurinovich) [inline Hometask4.playground/Contents.swift]: You can combine these lines of code into one. No need to declare a variable on one line and assign value in another

File: Hometask4.playground/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+//3.  Создать массив интересных фактов в формате (“название города/страны: факт”) из первоначального словаря, отсортировав их по длине строки по убыванию
+
+var countriesFactsArray = [String]()
+
+for (country, fact) in worldPlaceDictionary {
+    countriesFactsArray.append("\(country): \(fact.0)")
+    countriesFactsArray = countriesFactsArray.sorted(by: { $0.count > $1.count })
```
Comment (darya-gurinovich) [inline Hometask4.playground/Contents.swift]: It's better to move this line out of the cycle. This way it's performed in every iteration. One call after all elements are added is enough

File: Hometask4.playground/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+countriesFactsArray = worldPlaceDictionary.map { "\($0.key): \($0.value.0) "}.sorted(by: { $0.count > $1.count })
+
+//4.* Отсортировать страны/города по континенту. Отсортированные данные должны лежать в одном объекте (сами придумайте, в каком). В итоге, у вас должна быть возможность получить все страны/города переданному континенту. (Лучше тут сразу функцию реализовать, которая будет возвращать страны по переданному континенту. Чтобы я потом не задавалась, вопросом, как мне потом страны получить из вашего объекта)
+
+var sortedArrayByCountries = [String : [String]] ()
```
Comment (darya-gurinovich) [inline Hometask4.playground/Contents.swift]: Please, remove a space before `:` and `(`: `var sortedArrayByCountries = [String: [String]]()`

File: Hometask4.playground/Contents.swift [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+for key in worldPlaceDictionary {
+    sortedArrayByCountries[key.value.1, default: []] += [key.key]
+}
+print(sortedArrayByCountries["Europe"]  ?? 0) /* Honestly, I don't understand what this does at all [key.value.1, default: [] += [key.key]
+I found the answer to my question in the lecture, but it's still unclear to me. It seems to me that something with my head , simple things are difficult */
```
Comment (darya-gurinovich) [inline Hometask4.playground/Contents.swift:47]: `sortedArrayByCountries[key.value.1]` returns an optional array of `String`([String]?). We want to add a new element to this array. We need to unwrap this optional. If the value is `nil` we need to create an array for that key and add `key.key` into this array. If the value isn't nil, we need to add `key.key` into this array found array. But we can do it much easier using `default`. You need to set here a value that would be returned if the value is nil. So `sortedArrayByCountries[key.value.1, default: []]` always returns `[String]` and you can add an element here straight away without unwrapping and all following stuff

File: Hometask4.playground/Contents.swift [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+/*• Дан массив проверить, есть ли в массиве дубликаты.
+ 
+(Желательно использовать map, reduce, filter, sorted)*/
+
+let duplicatedNumbersArray = [1,2,43,5,6,3,5,6,6,3,3]
```
Comment (darya-gurinovich) [inline Hometask4.playground/Contents.swift]: Please, add a space after each `,`

**Review by darya-gurinovich — APPROVED** [review]
Good work! Thank you for commiting to do homework even when you don't feel the best. Please, take as much time as you need to get better. I hope I've clarified all unclear things

File: Hometask4.playground/Contents.swift (hunk 2 above)
Comment (Tryxman) [reply Hometask4.playground/Contents.swift]: added

File: Hometask4.playground/Contents.swift (hunk 3 above)
Comment (Tryxman) [reply Hometask4.playground/Contents.swift]: combined

File: Hometask4.playground/Contents.swift (hunk 5 above)
Comment (Tryxman) [reply Hometask4.playground/Contents.swift]: Done

File: Hometask4.playground/Contents.swift (hunk 7 above)
Comment (Tryxman) [reply Hometask4.playground/Contents.swift]: Added

File: Hometask4.playground/Contents.swift (hunk 6 above)
Comment (Tryxman) [reply Hometask4.playground/Contents.swift:47]: Hm , now it is clear for me , i need more practice with this thing to understand it at all

Comment (Tryxman) [reply Hometask4.playground/Contents.swift:47]: I found the "Swift Taskbook (language version 4.0)".
Can I use it in combination with codewars? Can I ask you to check my pull request with the tasks from "Taskbook"?

File: Hometask4.playground/Contents.swift (hunk 4 above)
Comment (Tryxman) [reply Hometask4.playground/Contents.swift]: Line moved out of cycle

File: Hometask4.playground/Contents.swift (hunk 6 above)
Comment (darya-gurinovich) [reply Hometask4.playground/Contents.swift:47]: Yeah, sure

**Review by darya-gurinovich — APPROVED** [review]

**Pull request merged**